src/main/java/net/mcreator/preferences/PreferencesManager.java

```diff
@@ -161,14 +161,17 @@ private static void convertOldPreferences() {
 		File file = UserFolderManager.getFileFromUserFolder("preferences");
 		JsonObject obj = gson.fromJson(FileIO.readFileToString(file), JsonObject.class);
 		PREFERENCES_REGISTRY.get(PreferencesData.CORE_PREFERENCES_KEY).forEach(entry -> {
-			JsonElement value = obj.get(entry.getSectionKey()).getAsJsonObject()
-					.get(entry.getID().replace("autoReloadTabs", "autoreloadTabs").replace("aaText", "aatext")
-							.replace("useMacOSMenuBar", "usemacOSMenuBar"));
+			// We check if the entry's section and the entry are defined to avoid crashes
+			if (obj.has(entry.getSectionKey()) && obj.get(entry.getSectionKey()).getAsJsonObject().has(entry.getID())) {

```

Comment:
Is this intentional that you don't perform replacement operations on the ID here?